engine/labels/labels.go

```diff
@@ -22,6 +22,11 @@ type LabelSelector[T Labeled] struct {
 	language gval.Language
 }
 
+// Helper function to refactor key value label manipulation
+func formatLabel(key, value string) string{
```

Comment:
Can we export this function so that we don't need all the other functions elsewhere? And maybe call it `AsString(key, value string)` or something. If value is empty, we should be removing the trailing "=" in the label.